src/cdk/accordion/accordion-item.ts

```diff
@@ -90,9 +90,9 @@ export class CdkAccordionItem implements OnDestroy {
 
   /** Whether the AccordionItem is disabled. */
   @Input()
-  get disabled() { return this._disabled; }
-  set disabled(disabled: any) { this._disabled = coerceBooleanProperty(disabled); }
-  private _disabled: boolean = false;
+  get disabled(): boolean { return this._disabled; }
+  set disabled(disabled: boolean) { this._disabled = coerceBooleanProperty(disabled); }
```

Comment:
The whole point of using coercion here is that you can pass in anything. The problem is that TS doesn't support different types for the getter and the setter.

Reply:
Yep, I know. And I'm aware this is a breaking change and I wouldn't expect it to be merged until you reach v11 stages.

Reply:
If we were actually requiring these to be boolean, the `coerceBooleanProperty` call becomes redundant. I think that changing this defeats the purpose and we'd have to do it across the entire codebase.

Reply:
You do actually use boolean for all boolean accessors, these here modified in this PR are outside the rule. It helps not only documentation, but the dev experience when accessing a _getter_, for example, you'd want a `boolean`, not `any`.

See these two examples for reference: https://github.com/angular/components/blob/cb8de6195eee87d7238fd6d3494b260b68ed5bd8/src/material/expansion/expansion-panel.ts#L107-L113
https://github.com/angular/components/blob/cfe37d18a67beddafd69ef95c2c605975e9b468f/src/cdk/drag-drop/directives/drop-list-group.ts#L36-L41